base/loading.jl

```diff
@@ -2,6 +2,52 @@
 
 # Base.require is the implementation for the `import` statement
 
+# Generic case-sensitive version of isfile
+function isfile_casesensitive_slow(path)
+    isfile(path) || return false
+    dir, filename = splitdir(path)
+    any(readdir(dir) .== filename)
+end
+
+isfile_casesensitive(path) = isfile_casesensitive_slow(path)
+
+@windows_only function isfile_casesensitive(path)
+    isfile(path) || return false
+    longpath(path) == path
+end
+
+@linux_only isfile_casesensitive(path) = isfile(path)
+
+const ATTR_BIT_MAP_COUNT=5
+const ATTR_CMN_NAME=1
+const BITMAPCOUNT = 1
+const COMMONATTR = 5
+const FSOPT_NOFOLLOW=1
+
+const attr_list = zeros(UInt8, 24)
+attr_list[BITMAPCOUNT] = ATTR_BIT_MAP_COUNT
+attr_list[COMMONATTR] = ATTR_CMN_NAME
+
+@osx_only function isfile_casesensitive(path)
+    isfile(path) || return false
+    buf_size = length(path) + 12
+    local canon_path
+    while true
+        buf = zeros(UInt8, buf_size)
```

Comment:
allocate `buf` outside the loop and use `resize!` in the loop?   Or does this have to be zeros?   If not, it would be clearer to use `Array(UInt8, buf_size)`